core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -103,10 +118,12 @@ public CallOptions withCompression(@Nullable String compressorName) {
    * @param deadlineNanoTime the deadline in the clock as per {@link System#nanoTime()}.
    *                         {@code null} for unsetting the deadline.
    */
+  @Deprecated
   public CallOptions withDeadlineNanoTime(@Nullable Long deadlineNanoTime) {
-    CallOptions newOptions = new CallOptions(this);
-    newOptions.deadlineNanoTime = deadlineNanoTime;
-    return newOptions;
+    Deadline deadline = deadlineNanoTime != null
+        ? Deadline.after(deadlineNanoTime - System.nanoTime(), TimeUnit.NANOSECONDS)
```

Comment:
is the potential for inaccuracy here a problem? Same question for `getDeadlineNanoTime()`.